### PR TITLE
Fixes #3240 : Renamed mockito bom artifact

### LINKED ISSUE #3240: Mockito bom missing artifact in maven central for java21
The following artifact is in the mockito bom but does not exist in maven central.  

```xml
      <dependency>
        <groupId>mockito</groupId>
        <artifactId>java21</artifactId>
        <version>5.9.0</version>
      </dependency>
```

This seems like it should not have been put in the bom.  This commit brought it in https://github.com/mockito/mockito/commit/a0cc22d73c184d0367f38c4d4d34de71c030c9b7.

### PR DESCRIPTION
**Pull request by PranotiPD**
## Description
- It makes a fix of 1 line in ***subprojects -> bom -> bom.gradle***
- changed ***!subproject.name.endsWith("-test")*** to ***!subproject.name.endsWith("java21-test")***
- By doing so now the artifact will be present in Maven central too.
## Checklist

 - [x] Read the [contributing guide](https://github.com/mockito/mockito/blob/main/.github/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [ ] If possible / relevant include an example in the description, that could help all readers
       including project members to get a better picture of the change
 - [x] Avoid other runtime dependencies
 - [x] Meaningful commit history ; intention is important please rebase your commit history so that each
       commit is meaningful and help the people that will explore a change in 2 years
 - [x] The pull request follows coding style
 - [ ] Mention `Fixes #<issue number>` in the description _if relevant_
 - [x] At least one commit should mention `Fixes #<issue number>` _if relevant_

### CODE REVIEW

File: subprojects/bom/bom.gradle [hunk 1]
```diff
@@ -15,7 +15,7 @@ dependencies {
         api(project.rootProject)
         project.rootProject.subprojects.forEach { subproject ->
             if (!subproject.name.endsWith("Test") &&
-                !subproject.name.endsWith("-test") &&
+                !subproject.name.endsWith("java21-test") &&
```
Comment (TWiStErRob) [inline subprojects/bom/bom.gradle:18]: The module name needs to change, not the logic.

Rename java21 to java21-test in settings.gradle and folder.

**Comment by TWiStErRob** [comment]
Tip: I think you can test this with `gradlew publishToMavenLocal`. Run it on `main` and this branch, and diff the generated files in `~/.m2/org/mockito`.

File: subprojects/bom/bom.gradle (hunk 1 above)
Comment (PranotiPD) [reply subprojects/bom/bom.gradle:18]: Ok. I'll make the required changes.

**Comment by TWiStErRob** [comment]
I'll let @TimvdLippe take it from here.

**Review by TimvdLippe — APPROVED** [review]
Thank you!

**Pull request merged**